app/views/thread_header.html

```diff
@@ -1,10 +1,13 @@
-<h3 class="question-cat">
-  {{ date_posted }} -
+<div class="QuestionMeta">
   {{ displayable_metadata.handset_type }}
-  {{ displayable_metadata.os_version }}
-  {% if (displayable_metadata.handset_type or displayable_metadata.os_version) and displayable_metadata.operator %}
-  , {% endif %}
-  {{ displayable_metadata.operator }}
+  {# Notice the dash below for whitespace control #}
+  {{ displayable_metadata.os_version }}{% if (displayable_metadata.handset_type or displayable_metadata.os_version) and displayable_metadata.operator -%}
```

Comment:
You need a dash at the beginning of the if to remove whitespace before that if.
You can put the if on its own line and it will work even if we only display `{{ displayable_metadata.handset_type }}`